diff -Naur ./newlib-4.4.0.20231231-origin/newlib/libc/include/sys/dirent.h ./newlib-4.4.0.20231231/newlib/libc/include/sys/dirent.h
--- ./newlib-4.4.0.20231231-origin/newlib/libc/include/sys/dirent.h	2023-12-31 18:00:18.000000000 +0100
+++ ./newlib-4.4.0.20231231/newlib/libc/include/sys/dirent.h	2024-04-01 13:48:15.907994623 +0200
@@ -1,13 +1,28 @@
-/* <dirent.h> includes <sys/dirent.h>, which is this file.  On a
-   system which supports <dirent.h>, this file is overridden by
-   dirent.h in the libc/sys/.../sys directory.  On a system which does
-   not support <dirent.h>, we will get this file which uses #error to force
-   an error.  */
-
 #ifdef __cplusplus
 extern "C" {
 #endif
-#error "<dirent.h> not supported"
+   
+#define MAXNAMLEN 4087
+     
+ struct dirent {
+     unsigned long d_reclen;
+     unsigned long d_ino;   
+     char d_name[MAXNAMLEN+1];
+ };
+   
+ typedef struct {
+     int dd_fd;  
+     char *dd_buf;
+     long dd_loc; 
+     long dd_size;
+     long dd_len; 
+ } DIR;
+ 
+ DIR *opendir (const char *);
+ struct dirent *readdir (DIR *);
+ int readdir_r(DIR *, struct dirent *, struct dirent **);
+ void rewinddir (DIR *);
+ int closedir (DIR *);  
 #ifdef __cplusplus
 }
 #endif
diff -Naur ./newlib-4.4.0.20231231-origin/newlib/libc/include/sys/resource.h ./newlib-4.4.0.20231231/newlib/libc/include/sys/resource.h
--- ./newlib-4.4.0.20231231-origin/newlib/libc/include/sys/resource.h	2023-12-31 18:00:18.000000000 +0100
+++ ./newlib-4.4.0.20231231/newlib/libc/include/sys/resource.h	2024-04-01 13:51:04.150358529 +0200
@@ -11,10 +11,16 @@
 #define	RUSAGE_CHILDREN	-1		/* terminated child processes */
 
 struct rusage {
-  	struct timeval ru_utime;	/* user time used */
+	struct timeval ru_utime;	/* user time used */
 	struct timeval ru_stime;	/* system time used */
 };
 
+#define PRIO_PROCESS 0
+#define PRIO_MIN -20
+#define PRIO_MAX 20
+
+int getpriority(int which, int who);
+int setpriority(int which, int who, int prio);
 int	getrusage (int, struct rusage*);
 
 #ifdef __cplusplus
diff -Naur ./newlib-4.4.0.20231231-origin/newlib/libc/include/sys/utime.h ./newlib-4.4.0.20231231/newlib/libc/include/sys/utime.h
--- ./newlib-4.4.0.20231231-origin/newlib/libc/include/sys/utime.h	2023-12-31 18:00:18.000000000 +0100
+++ ./newlib-4.4.0.20231231/newlib/libc/include/sys/utime.h	2024-04-01 14:03:47.175051208 +0200
@@ -15,6 +15,8 @@
   time_t modtime; 
 };
 
+extern int utime(const char *file, const struct utimbuf *buf);
+
 #ifdef __cplusplus
 };
 #endif
